### PR TITLE
Remind PR merge and new case failure 

### PR DESCRIPTION
**Pull request by KivenChen**
Added features:
- Comment PRs with failed test case with "issue closed" tag
- Comment PRs with failed test case that has not been recorded in the database

Refinement:
- Data types, ci report query fix, ...

This PR is the current up and running version started on Jan. 27th.

### CODE REVIEW

**Comment by KivenChen** [comment]
Note that GitHub related features (case-issue management and PR reminder) are currently switched off in the running version, specified by empty GitHub token (`-tk`)

**Comment by KivenChen** [comment]
PTAL @lilinghai

File: sync_ci/pkg/detect/detect.go [hunk 1]
```diff
@@ -97,19 +166,35 @@ func handleCasesIfIssueExists(cfg model.Config, recentCaseSet map[string]map[str
...
+			issueCase := model.CaseIssue{
+				IssueNo:   RETRIGGERED_CASE,
+				Repo:      repo,
+				IssueLink: sql.NullString{},
+				Case:      sql.NullString{caseName, true},
+				JobLink:   sql.NullString{joblinks[0], true},
+			}
+			issueCases = append(issueCases, &issueCase)
+		}else{
+			issueCase := model.CaseIssue{
```
Comment (AsterNighT) [inline sync_ci/pkg/detect/detect.go:189]: Duplicated code, an assignment to IssueNo will do

File: sync_ci/pkg/detect/detect.go [hunk 2]
```diff
@@ -32,58 +39,120 @@ func GetCasesFromPR(cfg model.Config, startTime time.Time, inspectStartTime time
...
 
 	recentRows, err := cidb.Raw(model.GetCICaseSql, formatT(startTime), formatT(now)).Rows()
 	if err != nil {
 		return nil, err
 	}
-	DupRecentCaseSet := map[string]map[string][]string{}
-	allRecentCases := getDuplicatesFromHistory(recentRows, caseSet, DupRecentCaseSet)
+	DupRecentCaseSet, allRecentCases := getDuplicatesFromHistory(recentRows, caseLinks, casePrSet)
+
+	dupCaseStr, err := json.Marshal(DupRecentCaseSet)
```
Comment (AsterNighT) [inline sync_ci/pkg/detect/detect.go:50]: handle the err here

File: sync_ci/pkg/detect/detect.go [hunk 3]
```diff
@@ -32,58 +39,120 @@ func GetCasesFromPR(cfg model.Config, startTime time.Time, inspectStartTime time
...
+	// create alerts and filter requiredCases
+	rowsToRemindPr, err := cidb.Raw(model.GetCICaseSql, formatT(startTime), formatT(now)).Rows()
+	if err != nil {
+		return nil, err
+	}
+	casesToRemind, _ := getHistoryCases(rowsToRemindPr, baselink)
+	handlePrReminder(cfg, dbGithub, dbIssueCase, casesToRemind, test)
+
 	// assumed `cases` param has no reps
 	_, err = handleCasesIfIssueExists(cfg, allRecentCases, dbIssueCase, dbGithub, true, test)
```
Comment (AsterNighT) [inline sync_ci/pkg/detect/detect.go:66]: same for this one

File: sync_ci/pkg/detect/detect.go [hunk 4]
```diff
@@ -32,58 +39,120 @@ func GetCasesFromPR(cfg model.Config, startTime time.Time, inspectStartTime time
...
+				log.S().Error("failed to check existing [case, repo]: ", c, repo)
+				continue
+			}
+			if !existedCases.Next() {
+				// no issue found. do nothing
+				continue
+			} else {
+				// check time
+				var issueNumStr string
+				err = existedCases.Scan(&issueNumStr)
```
Comment (AsterNighT) [inline sync_ci/pkg/detect/detect.go:99]: ditto

File: sync_ci/pkg/parser/ci_log.go [hunk 5]
```diff
@@ -101,6 +101,17 @@ func refineParseRes(res map[string][]string) {
 			delete(res, v)
 		}
 	}
+
+	// The input will only fall on one category according to ranking
```
Comment (AsterNighT) [inline sync_ci/pkg/parser/ci_log.go:105]: I'm a bit confused. Wouldn't there be some possibility for one job having two sorts of errors?

**Pull request merged**